UltraStar Play/Assets/src/model/song/builder/SongMetaBuilder.cs

```diff
@@ -63,15 +64,15 @@ public static SongMeta ParseFile(string path, Encoding enc = null)
                 {
                     requiredFields[tag] = val;
                 }
-                else if (tag.StartsWith("p", StringComparison.Ordinal) && tag.Length > 1)
+                else if (tag.StartsWith("p", StringComparison.Ordinal) && tag.Length > 1 && Regex.IsMatch(tag, @"^p\d+$"))
```

Comment:
Please change such often called Regex checks to a compiled regex, which is instantiated once and then reused.
(for example Regex myRegex = new Regex(pattern, RegexOptions.Singleline | RegexOptions.Compiled);